sklearn/neighbors/_nca.py

```diff
@@ -195,6 +197,7 @@ def __init__(
         self.callback = callback
         self.verbose = verbose
         self.random_state = random_state
+        self._n_features_out = n_components
```

Comment:
We can not set private attributes in `__init__`. We have common checks that test for this behavior. You can see if on the [CI failing tests](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=36979&view=logs&j=dde5042c-7464-5d47-9507-31bdd2ee0a3a&t=4bd2dad8-62b3-5bf9-08a5-a9880c530c94).

Reply:
Now passes the tests (at least on my machine)